File: 230.kth-smallest-element-in-a-bst.py
#
# @lc app=leetcode id=230 lang=python3
#
# [230] Kth Smallest Element in a BST
#


# @lc code=start
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        preorder = self.getPreorder(root)
        return preorder[k - 1]

    def getPreorder(self, root):
        if root is None:
            return []
        return self.getPreorder(root.left) + [root.val] + self.getPreorder(root.right)


# @lc code=end

File: 230.kth-smallest-element-in-a-bst.go
/*
 * @lc app=leetcode id=230 lang=golang
 *
 * [230] Kth Smallest Element in a BST
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func kthSmallest(root *TreeNode, k int) int {
	vals := inorder(root)
	return vals[k-1]
}

func inorder(root *TreeNode) []int {
	if root == nil {
		return []int{}
	}
	return append(append(inorder(root.Left), root.Val), inorder(root.Right)...)
}

// @lc code=end

